### PR TITLE
SC read channels

### PR DESCRIPTION
**Pull request by scamp925**
- [x] IN ORDER TO SEE ANYTHING, YOU NEED TO MAKE SURE THE SAMPLE DATA IN FIREBASE HAS YOUR UID AS A 'UID' VALUE
- [x]  npm run dev
- [x] Index.js => currently, I have just a blank channel render on index. I am thinking when we have merge data on channel and posts, then I suggest this page by our default "general" channel
- [x] On the sidebar, click 'Channels' then all current channels associated with your uid will render in the dropdown as well as a '+ Add a Channel' option (nothing to do there yet)
- [x] Once you click a specific channel, in the channel's navbar you will see the channel's name as well as the general structure of a channel's page
- [x] Text Input box will be on the bottom of the page with the help of a little css
- [x] I have a scroll option in the css, but it won't populate until needed

### CODE REVIEW

File: components/Channel.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    </div>
+  );
+}
+
+Channel.propTypes = {
+  channelObj: PropTypes.shape({
+    channelName: PropTypes.string,
+    firebaseKey: PropTypes.string,
+    public: PropTypes.bool,
+  }).isRequired,
```
Comment (Massacre-Saint) [inline components/Channel.js:30]: If no channels exist, we need to have defaultProps. Is currently throwing error when empty.

**Review by Massacre-Saint — CHANGES_REQUESTED** [review]

File: components/Sidebar.js [hunk 2]
```diff
@@ -13,12 +23,14 @@ export default function Sidebar() {
...
-          <div>Channel 2</div>
-          <div>Channel 3</div>
-          <div>Add channels</div>
+        <div className="dropdown">
+          <button className="btn btn-dark btn-sm dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false">
+            Channels
+          </button>
+          <ul className="dropdown-menu">
+            {channelList.map((channel) => <li><Link className="dropdown-item" href={`/channel/${channel.firebaseKey}`}>{channel.channelName}</Link></li>)}
+            <li><Link className="dropdown-item" href="/">+ Add channel</Link></li>
```
Comment (Massacre-Saint) [inline components/Sidebar.js:32]: Will need to add a key={firebasekey} for the parent component ? that will be repeating

**Review by rochelle-rossman — APPROVED** [review]

**Review by kmchandler — APPROVED** [review]
!!

File: components/Channel.js (hunk 1 above)
Comment (scamp925) [reply components/Channel.js:30]: Updated with new defaultProps

**Review by Massacre-Saint — APPROVED** [review]

**Pull request merged**